### PR TITLE
Doc for cased wellbore problem

### PR DESCRIPTION
**Pull request by sytuannguyen**

### CODE REVIEW

File: src/docs/sphinx/advancedExamples/elasticCasedWellbore/Index.rst [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+.. literalinclude:: src/coreComponents/physicsSolvers/solidMechanics/benchmarks/CasedElasticWellbore.xml
+  :language: xml
+  :start-after: <!-- SPHINX_Material -->
+  :end-before: <!-- SPHINX_MaterialEnd -->
+
+
+FieldSpecifications: define boundary conditions
+--------------------------------------------------------------------
+
+Far-field boundary are fixed in displacements. The normal traction on the inner face of the casing is defined by ``Traction`` field specification. The nodeset generated by the internal wellbore generator for the inner face is ``rneg``. The traction type for defining a casing test pressure that is normal to the casing inner face is ``normal``. The negative sign of the scale value is because of the negative sign convention for compressional stress.
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/elasticCasedWellbore/Index.rst:82]:
```suggestion
Far-field boundary are subjected to roller constraints. The normal traction on the inner face of the casing is defined by ``Traction`` field specification. The nodeset generated by the internal wellbore generator for this face is named as ``rneg``. The traction type is ``normal`` to mimic a casing test pressure that is applied normal to the casing inner face . The negative sign of the scale is attributed to the negative sign convention for compressive stress in GEOSX.
```

**Review by jhuang2601 — COMMENTED** [review]
Good work!

File: src/docs/sphinx/advancedExamples/elasticCasedWellbore/Index.rst [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+Cased Elastic Wellbore Problem without Fluid Flow
+####################################################
+
+------------------------------------------------------------------
+Problem description
+------------------------------------------------------------------
+
+This example uses the solid mechanics solver to handle a cased wellbore problem subjected to a pressure test. The completed wellbore is composed of a steel casing, a cement sheath and rock formation. Isotropic linear elastic behavior is assumed for all the three materials. No separation is allowed for the casing-cement and cement-rock contact interfaces.
+
+Analytical results of the radial and hoop stresses, :math:`\sigma_{rr}` and :math:`\sigma_{\theta\theta}`, in casing, cement sheath and rock are expressed as:
```
Comment (sytuannguyen) [inline src/docs/sphinx/advancedExamples/elasticCasedWellbore/Index.rst:14]:
```suggestion
Analytical results of the radial and hoop stresses, :math:`\sigma_{rr}` and :math:`\sigma_{\theta\theta}`, in casing, cement sheath and rock are expressed as (Hervé and Zaoui, 1995):
```

File: src/docs/sphinx/advancedExamples/elasticCasedWellbore/Index.rst [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+This example uses the solid mechanic solver for solving a cased wellbore problem subjected to a pressure test. The problem geometry is composed a steel casing, a cement sheath and rock formation. Isotropic linear elastic behavior is assumed for all the three materials. The casing-cement and cement-rock contact interfaces are supposed to be perfect.
+
+Analytical results of radial and hoop stresses, :math:`\sigma_{rr}` and :math:`\sigma_{\theta\theta}`, in casing, cement sheath and rock can be expressed as:
+
+.. math::
+   \sigma_{rr} = ( 2.0 * \lambda + 2.0 * G ) * A - \frac{2.0 * G * B}{r^2}
+   \sigma_{\theta\theta} = ( 2.0 * \lambda + 2.0 * G ) * A + \frac{2.0 * G * B}{r^2}
+
+where :math:`\lambda` and :math:`G` are the Lamé moduli, :math:`r` is the radial coordinate, :math:`A` and :math:`B` are piecewise constants that can be obtained by solving the boundary and interface conditions as detailed in the benchmark script.
+
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/elasticCasedWellbore/Index.rst:21]: Added the reference of Hervé and Zaoui (1995) where this analytical results can be found.

**Review by jhuang2601 — APPROVED** [review]

**Review by joshua-white — APPROVED** [review]

**Pull request merged**